### DISCUSSION TITLE
Edit 'Note Source'

### DISCUSSION THREAD

**Discussion by Meroyanae** [TriliumNext/Trilium#4449] [Q&A] [answered]
Hi, is it possible to edit the notes' sources, e.g. to rewrite HTML or CSS?

Edit : To clarify my question, is it possible to change the sources of the notes directly in the "Note source" menu? I can't do it.

I can make changes, but to do so I have to export, make my changes and then import. But it's a "cumbersome" operation.

**Comment by zerebos** [comment]
No, Trilium is meant as a WYSIWYG note taking application rather than building custom notes with custom html.

**Reply by Meroyanae** [reply]
Ok, thanks for your quick reply.

**Comment by rodrigoalexandrecosta** [comment, marked as answer]
Well, if you really want (or need) to deal with HTML directly, perhaps you could create a Code note of HTML type and render this note using a Render note. From what I've seen, this is usually done to render specific pages/styles for scripts, and keep in mind that your content (your text and so on) would also need to be in the HTML note (perhaps not the best way to edit text), but can be done.

**Reply by Meroyanae** [reply]
It's an interesting way of doing things. I wanted to modify the HTML code to reorganize some tables. I'm going to try this method to see if it's not too complex to implement for tables.
Thanks @rodrigoalexandrecosta

**Reply by Meroyanae** [reply]
In my case, it works perfectly! Thanks again @rodrigoalexandrecosta !

**Comment by coleman-rik** [comment]
I realize that I'm a bit late to the party on this issue, but if this is the case, the application doesn't seem fit for purpose.

I have an existing note that I want to add some more text to.  Trillium will not let me add text without creating a space between the existing text and the new text.  Looking at the note's source I can see the problem is that Trillium is insisting on creating a new set of paragraph tags around the additional text instead of adding the new text to the existing paragraph.
It's a trivial fix if I was allowed to edit the source for the note, but since I am not, I will have to copy the note text to an exterior text editor, make my changes there and then copy them back into a new note or delete all of the existing text in this note and replace it with the updated text.

Having a WYSISYG editor is nice in theory, if it handles edits in a rational way.  Trillium obviously doesn't.

Without having more control over the editing process, Trillium is outmatched by a simple collection of documents in a folder and a text editor capabile of opening all of the files in a folder like Kate or Microsoft Code.

**Reply by eliandoran** [reply]
> I have an existing note that I want to add some more text to. Trillium will not let me add text without creating a space between the existing text and the new text. 

That's generally how most WYSIWYG editors do, they create paragraphs when pressing enter. To avoid that, you can press Shift+Enter which will insert an inline break.

> It's a trivial fix if I was allowed to edit the source for the note

You can edit the source, but it's slightly more difficult. You have to change the note type from "Text" to "Plain text" (or HTML), make your change and then go back to the "Text" note type.

> Without having more control over the editing process, Trillium is outmatched by a simple collection of documents in a folder and a text editor capabile of opening all of the files in a folder like Kate or Microsoft Code.

If you need a Kate-like or VS Code-like editor, try the Code notes (right click on a note -> Insert child note -> Code).

**Reply by coleman-rik** [reply]
Are either of these solutions incorporated into the Trillium documentation?

Namely the requirement to use Shift-Enter to **not** accidentilly create another paragraph when trying to edit an existing note, and
having to switch the type of note, from Text to something else (ignoring the ominous warning message presented when editing an existing note) making changes and switching it back?

I couldn't find either anywhere in my search.

Thanks.

**Reply by eliandoran** [reply]
> to use Shift-Enter to not accidentilly create another paragraph

https://docs.triliumnotes.org/User%20Guide/User%20Guide/Note%20Types/Text/Keyboard%20shortcuts.html?h=shift#content-editing

> having to switch the type of note, from Text to something else

https://docs.triliumnotes.org/User%20Guide/User%20Guide/Advanced%20Usage/Note%20source.html

**Comment by eliandoran** [comment]
As discussed previously, there are two types of changing the note source:

1. Temporarily switching the note type from "Text" or any other type really to a code note ("Plain text"), making the change and reverting the note type back.
2. Using a render HTML as specified by @rodrigoalexandrecosta .
